dm/dm/worker/source_worker.go

```diff
@@ -455,7 +456,7 @@ func (w *SourceWorker) EnableHandleSubtasks() error {
 
 	// we get the newest subtask stages directly which will omit the subtask stage PUT/DELETE event
 	// because triggering these events is useless now
-	subTaskStages, subTaskCfgM, revSubTask, err := w.fetchSubTasksAndAdjust()
+	subTaskStages, validatorStages, subTaskCfgM, revSubTask, err := w.fetchSubTasksAndAdjust()
```

Comment:
I found this is the only usage of `validatorStages` from `w.fetchSubTasksAndAdjust()`, meanwhile we have called `getExpectValidatorStage` at two place. Can we only use `getExpectValidatorStage` and not modify `w.fetchSubTasksAndAdjust()` (for example `w.StartSubTask` will call `getExpectValidatorStage` inside)?

Reply:
`fetchSubTasksAndAdjust` is fetching in a transaction in batch. for `getExpectValidatorStage`, it's getting for a single task